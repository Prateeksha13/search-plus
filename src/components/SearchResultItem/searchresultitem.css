.search-result-item {
  display: flex;
  justify-content: space-between;
}

.search-result-item .store-information {
  flex: 1;
  text-align: left;
}

.search-result-item .stock-information {
  flex: 1;
  text-align: right;
}

.search-result-item .content-green {
  color: var(--text-green);
}

.search-result-item .content-red {
  color: var(--text-red);
}
