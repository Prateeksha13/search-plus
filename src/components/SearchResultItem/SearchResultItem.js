import React from "react";

import "./searchresultitem.css";

const SearchResultItem = ({
  itemNumber,
  locality,
  storeName,
  storeAddress,
  isProductAvailable,
  stockQty,
}) => {
  return (
    <div className="search-result-item">
      <div className="serial-number">{itemNumber}</div>
      <div className="store-information">
        <h4>
          {locality}, {storeName}
        </h4>
        <address>
          <p>{storeAddress[0]}</p>
          <p>{storeAddress[1]}</p>
        </address>
      </div>
      <div className="stock-information">
        <div>
          Available:{" "}
          {isProductAvailable && <span className="content-green">Yes</span>}
          {!isProductAvailable && <span className="content-red">No</span>}
        </div>
        <div>
          {isProductAvailable && (
            <span className="content-green">In Stock: {stockQty}</span>
          )}
          {!isProductAvailable && (
            <span className="content-red">Not In Stock</span>
          )}
        </div>
      </div>
    </div>
  );
};

export default SearchResultItem;
