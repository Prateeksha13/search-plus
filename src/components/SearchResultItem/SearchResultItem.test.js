import React from "react";
import renderer from "react-test-renderer";
import SearchResultItem from "./SearchResultItem";

test("renders result item", () => {
  const component = renderer.create(
    <SearchResultItem
      itemNumber={1}
      locality={"Toledo"}
      storeName={"Franklin Park Mall"}
      storeAddress={["5001 Monroe Street, Toledo, OH 43623", "(419) 473-4250"]}
      stockQty={20}
      isProductAvailable={true}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
