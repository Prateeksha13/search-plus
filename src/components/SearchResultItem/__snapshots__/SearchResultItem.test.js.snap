// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders result item 1`] = `
<div
  className="search-result-item"
>
  <div
    className="serial-number"
  >
    1
    .
  </div>
  <div
    className="store-information"
  >
    <h4
      className="store-name"
      role="heading"
    >
      Toledo
      ,
       
      <span
        className="highlighted-store-name"
      >
        Franklin Park Mall
      </span>
    </h4>
    <address
      className="store-address"
    >
      <p>
        5001 Monroe Street, Toledo, OH 43623
      </p>
      <p>
        (419) 473-4250
      </p>
    </address>
  </div>
  <div
    className="stock-information"
  >
    <div
      className="stock-detail"
    >
      Available:
       
      <span
        className="content-green"
      >
        Yes
      </span>
    </div>
    <div
      className="stock-detail"
    >
      <span
        className="content-green"
      >
        In Stock: 
        20
      </span>
    </div>
  </div>
</div>
`;
